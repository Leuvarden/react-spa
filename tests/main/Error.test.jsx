import React from 'react';
import ErrorBoundary from './../../src/components/Error.js';

/* eslint-disable no-undef */
describe ('ErrorBoundary', () => {

    const defaultProps = {
        showOnError: <h1>Something went wrong</h1>
    };

    it('renders as expected without error', () => {
        const item = shallow(
            <ErrorBoundary/>
        );
    
        expect(item).toMatchSnapshot();
    });

    it('renders as expected with error', () => {
        const item = shallow(
            <ErrorBoundary {...defaultProps}>huh</ErrorBoundary>
        );

        const itemUpdate = item.setState({hasError: true});
    
        expect(itemUpdate).toMatchSnapshot();
    });

});