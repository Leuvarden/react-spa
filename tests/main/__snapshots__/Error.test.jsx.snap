// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary renders as expected with error 1`] = `
<h1>
  Something went wrong
</h1>
`;

exports[`ErrorBoundary renders as expected without error 1`] = `""`;
